### ISSUE TITLE
malloc.h missing

### ISSUE THREAD

**Issue by dedenker** [michalivne/BGSubtraction#1]
Can this LIB (MALLOC.H) be replace with STDLIB.H I did not go over the code more.
But I am trying to build this on a pretty basic Ubuntu 14.04 x86 image.

*[Reference](http://stackoverflow.com/questions/12973311/difference-between-stdlib-h-and-malloc-h)

**Comment by michalivne** [comment]
I'm not sure. What kind of error do you get?

**Comment by dedenker** [comment]
When I try to run setup.py (sudo python setup.py install), I get a compilation error where it can not find MALLOC.H.
But when I search the web for the correct package that will provide this, everywhere I read that MALLOC.H is best replaced by STDLIB.H.
So I see that you last commit was a year ago, I was wondering if the source perhaps needs an update....?

**Comment by michalivne** [comment]
It is the wrong building  method. setup.py is not the installation script.
Try using build_all instead. It uses CMake so it will fail with errors for
missing libraries.

I will add a readme to explain that.
​

Micha Livne

**Comment by dedenker** [comment]
I see, but I still get an error, this:
./build_all -DOpenCV_DIR=/media/tijdelijk/Downloads/opencv-2.4.11/build/
-- The C compiler identification is GNU 4.8.4
-- The CXX compiler identification is GNU 4.8.4
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Boost version: 1.54.0
-- Boost version: 1.54.0
-- Found the following Boost libraries:
--   system
-- Found SWIG: /usr/bin/swig2.0 (found version "2.0.11") 
-- Found PythonLibs: /usr/lib/i386-linux-gnu/libpython2.7.so (found version "2.7.6") 
-- Found PythonInterp: /usr/bin/python (found version "2.7.6") 
-- Found NumPy: 1.8.2  
-- Configuring done
-- Generating done
-- Build files have been written to: /media/tijdelijk/Downloads/bgs-python/src/cpp/build
Scanning dependencies of target bgs
[  1%] [  2%] [  3%] [  4%] [  5%] Building CXX object bgs_library/CMakeFiles/bgs.dir/PreProcessor.cpp.o
Building CXX object bgs_library/CMakeFiles/bgs.dir/VideoAnalysis.cpp.o
Building CXX object bgs_library/CMakeFiles/bgs.dir/FrameProcessor.cpp.o
Building CXX object bgs_library/CMakeFiles/bgs.dir/VideoCapture.cpp.o
Building CXX object bgs_library/CMakeFiles/bgs.dir/package_bgs/jmo/CMultiLayerBGS.cpp.o
[  7%] [  8%] In file included from /media/tijdelijk/Downloads/bgs-python/src/cpp/bgs_library/FrameProcessor.cpp:17:0:
/media/tijdelijk/Downloads/bgs-python/src/cpp/bgs_library/FrameProcessor.h:18:0: warning: ignoring #pragma warning  [-Wunknown-pragmas]
 #pragma warning(disable : 4482)
 ^
In file included from /media/tijdelijk/Downloads/bgs-python/src/cpp/bgs_library/FrameProcessor.h:20:0,
                 from /media/tijdelijk/Downloads/bgs-python/src/cpp/bgs_library/FrameProcessor.cpp:17:
/media/tijdelijk/Downloads/bgs-python/src/cpp/bgs_library/IFrameProcessor.h:19:16: fatal error: cv.h: No such file or directory
 #include <cv.h>
                ^
compilation terminated.
Building CXX object bgs_library/CMakeFiles/bgs.dir/package_bgs/jmo/blob.cpp.o
Building CXX object bgs_library/CMakeFiles/bgs.dir/package_bgs/jmo/BlobExtraction.cpp.o
In file included from /media/tijdelijk/Downloads/bgs-python/src/cpp/bgs_library/package_bgs/jmo/CMultiLayerBGS.h:57:0,
                 from /media/tijdelijk/Downloads/bgs-python/src/cpp/bgs_library/package_bgs/jmo/CMultiLayerBGS.cpp:47:
/media/tijdelijk/Downloads/bgs-python/src/cpp/bgs_library/package_bgs/jmo/LocalBinaryPattern.h:50:16: fatal error: cv.h: No such file or directory
 #include <cv.h>
                ^
compilation terminated.
[  9%] Building CXX object bgs_library/CMakeFiles/bgs.dir/package_bgs/jmo/BlobResult.cpp.o
make[2]: **\* [bgs_library/CMakeFiles/bgs.dir/FrameProcessor.cpp.o] Error 1
make[2]: **\* Waiting for unfinished jobs....
make[2]: **\* [bgs_library/CMakeFiles/bgs.dir/package_bgs/jmo/CMultiLayerBGS.cpp.o] Error 1
In file included from /media/tijdelijk/Downloads/bgs-python/src/cpp/bgs_library/package_bgs/jmo/blob.cpp:60:0:
/media/tijdelijk/Downloads/bgs-python/src/cpp/bgs_library/package_bgs/jmo/blob.h:63:20: fatal error: cxcore.h: No such file or directory
 #include "cxcore.h"
                    ^
compilation terminated.
In file included from /media/tijdelijk/Downloads/bgs-python/src/cpp/bgs_library/package_bgs/jmo/BlobExtraction.cpp:71:0:
/media/tijdelijk/Downloads/bgs-python/src/cpp/bgs_library/package_bgs/jmo/BlobResult.h:62:20: fatal error: cxcore.h: No such file or directory
 #include "cxcore.h"
                    ^
compilation terminated.
make[2]: **\* [bgs_library/CMakeFiles/bgs.dir/package_bgs/jmo/blob.cpp.o] Error 1
make[2]: **\* [bgs_library/CMakeFiles/bgs.dir/package_bgs/jmo/BlobExtraction.cpp.o] Error 1
In file included from /media/tijdelijk/Downloads/bgs-python/src/cpp/bgs_library/package_bgs/jmo/BlobResult.cpp:56:0:
/media/tijdelijk/Downloads/bgs-python/src/cpp/bgs_library/package_bgs/jmo/BlobResult.h:62:20: fatal error: cxcore.h: No such file or directory
 #include "cxcore.h"
                    ^
compilation terminated.
In file included from /media/tijdelijk/Downloads/bgs-python/src/cpp/bgs_library/PreProcessor.cpp:17:0:
/media/tijdelijk/Downloads/bgs-python/src/cpp/bgs_library/PreProcessor.h:20:16: fatal error: cv.h: No such file or directory
 #include <cv.h>
                ^
compilation terminated.
In file included from /media/tijdelijk/Downloads/bgs-python/src/cpp/bgs_library/VideoCapture.cpp:17:0:
/media/tijdelijk/Downloads/bgs-python/src/cpp/bgs_library/VideoCapture.h:20:16: fatal error: cv.h: No such file or directory
 #include <cv.h>
                ^
compilation terminated.
In file included from /media/tijdelijk/Downloads/bgs-python/src/cpp/bgs_library/VideoAnalysis.h:22:0,
                 from /media/tijdelijk/Downloads/bgs-python/src/cpp/bgs_library/VideoAnalysis.cpp:17:
/media/tijdelijk/Downloads/bgs-python/src/cpp/bgs_library/VideoCapture.h:20:16: fatal error: cv.h: No such file or directory
 #include <cv.h>
                ^
compilation terminated.
make[2]: **\* [bgs_library/CMakeFiles/bgs.dir/package_bgs/jmo/BlobResult.cpp.o] Error 1
make[2]: **\* [bgs_library/CMakeFiles/bgs.dir/VideoCapture.cpp.o] Error 1
make[2]: **\* [bgs_library/CMakeFiles/bgs.dir/PreProcessor.cpp.o] Error 1
make[2]: **\* [bgs_library/CMakeFiles/bgs.dir/VideoAnalysis.cpp.o] Error 1
make[1]: **\* [bgs_library/CMakeFiles/bgs.dir/all] Error 2
make: **\* [all] Error 2

Do I use a wrong version of OpenCV(2.4.11)?

**Comment by michalivne** [comment]
I don't think you can use any flags. You should define the OpenCV directory as an environment variable. 

Micha Livne
